charts/iter8lib/templates/_k-spec-secret.tpl

```diff
@@ -7,5 +8,13 @@ metadata:
     iter8.tools/revision: {{ .Release.Revision | quote }}
 stringData:
   experiment.yaml: |
-{{- include "experiment" . | indent 4 }}
+{{ include "experiment" . | indent 4 }}
+{{- if .Values.providers }}
+  metrics.yaml: |
+{{- range .Values.providers }}
+{{- if eq . "istio"}}
```

Comment:
@sriumcp I was able to implement what you wanted, which was to dynamically construct the name of the template. 

This doesn't seem to be a common pattern. [This](https://stackoverflow.com/questions/20716726/call-other-templates-with-dynamic-name) was all I could find on the subject and it seems to be a frowned upon practice.

Reply:
We don't need to worry about `iter8lib` at all ... at the moment. Especially, changes to secret, cronjob, etc. can be handled in a separate PR.